src/journal/journald-file.c

```diff
@@ -91,8 +92,14 @@ static int journald_file_entry_array_punch_hole(JournalFile *f, uint64_t p, uint
                 return 0;
         }
 
-        if (fallocate(f->fd, FALLOC_FL_PUNCH_HOLE | FALLOC_FL_KEEP_SIZE, offset, sz) < 0)
+        if (fallocate(f->fd, FALLOC_FL_PUNCH_HOLE | FALLOC_FL_KEEP_SIZE, offset, sz) < 0) {
+                if (ERRNO_IS_NOT_SUPPORTED(errno)) {
+                        log_debug("Hole punching not supported by backing file system, skipping.");
+                        return -EOPNOTSUPP; /* Make recognizable */
```

Comment:
`SYNTHETIC_ERRNO()`?

Reply:
I thought about that, but it's not as obvious where to place the `/* Make recognizable */` comment then.

Reply:
OK. Let's keep this as is.